### PR TITLE
Added selection sort

### PR DESCRIPTION
**Pull request by bhavyakaria**
I have added the code for the selection sort and also provided test cases.

### CODE REVIEW

File: src/main/kotlin/sort/SelectionSort.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        // finding minimum in unsorted array
+        var idx = i
+        for(j in i+1..length) {
+            if(array[j] < array[idx]) {
+                idx = j
+            }
+        }
+
+        // swap elements
+        swapElements(array,i,idx)
```
Comment (varunu28) [inline src/main/kotlin/sort/SelectionSort.kt]: Please update it according to coding standards.
```
swapElements(array, i, idx)
```

File: src/test/kotlin/sort/SelectionSortTest.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        Assert.assertArrayEquals(array, arrayOf(1, 2, 3, 4, 8))
+    }
+
+    @Test
+    fun testSelectionSort2() {
+        val array = arrayOf(20,5,16,-1,6)
+        selectionSort(array)
+
+        Assert.assertArrayEquals(array, arrayOf(-1,5,6,16,20))
```
Comment (varunu28) [inline src/test/kotlin/sort/SelectionSortTest.kt]: Same here. Please keep the coding standard consistent.

**Comment by varunu28** [comment]
@bhavyakaria Thanks for the contribution. I have added a few comments regarding some minor changes.

**Comment by bhavyakaria** [comment]
@varunu28 Thanks for the review, have made the required changes.

File: src/test/kotlin/sort/SelectionSortTest.kt [hunk 3]
```diff
@@ -10,22 +10,22 @@ class SelectionSortTest {
...
         selectionSort(array)
 
-        Assert.assertArrayEquals(array, arrayOf(-1,5,6,16,20))
+        assertArrayEquals(array, arrayOf(-1, 5, 6, 16, 20))
     }
 
     @Test
     fun testSelectionSort3() {
         val array = arrayOf("A", "D", "E", "C", "B")
         bubbleSort(array)
```
Comment (varunu28) [inline src/test/kotlin/sort/SelectionSortTest.kt]: Why are we calling bubbleSort here?

Comment (bhavyakaria) [reply src/test/kotlin/sort/SelectionSortTest.kt]: Rectified my mistake.

**Comment by varunu28** [comment]
@bhavyakaria Thanks a ton for the PR. Merging it

**Pull request merged**